blockcerts/signaturetracker.go

```diff
@@ -0,0 +1,139 @@
+package blockcerts
+
+import (
+    "context"
+    "fmt"
+    "sync"
+
+    certtypes "github.com/spacemeshos/go-spacemesh/blockcerts/types"
+    "github.com/spacemeshos/go-spacemesh/common/types"
+    "github.com/spacemeshos/go-spacemesh/log"
+)
+
+type SigCacher interface {
+    CacheBlockSignature(ctx context.Context,
+        blockSignature certtypes.BlockSignatureMsg)
+}
+
+type sigCache struct {
+    logger             log.Logger
+    blockSigsByLayer   sync.Map // entries are written once but read many times
+    cacheBoundary      types.LayerID
+    cacheBoundaryMutex sync.RWMutex
+    signaturesRequired int
```

Comment:
i don't understand the design of sigCache and blockSignatureTracker here.

you basically need a object, say certificateTracker that holds a `sigs map[blockID][]signatures`, and when the gossip handler call CacheBlockSignature:
```
sigs[bid] = append(sigs[bid], sig)
if len(sigs) >= signaturesRequired {
  t.completedCerts <- generateCert(sigs[bid])
}
delete(sigs, bid)
```
what are `blockSigsByLayer` and `cacheBoundary` for?